src/drivers/ports/esp32/can_port.i

```diff
@@ -220,6 +300,49 @@ int can_port_init(struct can_driver_t *self_p,
     return (0);
 }
 
+/**
+ *  Wait some time for bus is idle.
+ *
+ * @param[in] self_p Driver object to work with.
+ *
+ * @return zero(0) if success or (-ENETDOWN) is case of timeout.
+ *
+ * CAN-module can be in state waiting for bus idle. At reset it waits for
+ * 11 consecutive recessive bits, so we will wait for at least 170 bit-periods
+ * (128*1.2+11 = 164, where 128 - is maximum CAN-frame length, 1.2 - bit stuffing koeff).
+ * In transition from bus-off to bus-on state it waits by default for 128 periods of
+ * 11 consecutive recessive bits but this number 128 is readed from TXERR register. So
+ * this function will work in all situations if TXERR is cleared.
+ */
+static int wait_for_bus_idle(struct can_driver_t *self_p)
+{
+    volatile struct esp32_can_t *regs_p;
+    struct time_t now, stop, delta;
+    uint8_t still_waiting_for_bus_idle;
+
+    regs_p = self_p->dev_p->regs_p;
+
+    still_waiting_for_bus_idle = 1;
+    time_get(&now);
+    delta.seconds = 0;
+    delta.nanoseconds = 17000L * 1000;    // 17000 = 170 * 100: 170 cycles of bit-period @ 10 KBPS (100 us per bit)
+    time_add(&stop, &now, &delta);
+    while ( (now.seconds < stop.seconds) || (now.nanoseconds < stop.nanoseconds) ) {
+
+        still_waiting_for_bus_idle = is_waiting_for_bus_idle(regs_p->STATUS);
+        if (! still_waiting_for_bus_idle) {
+            break;
+        }
+
+        time_get(&now);
+    }
+    if (still_waiting_for_bus_idle) {
+        return (-ENETDOWN);
+    } else {
+        return (0);
+    }
+}
+
```

Comment:
Change to the code below, and move the function directly after ``is_waiting_for_bus_idle()``. Also, make the function that ``regs_p`` as arguement instead of ``self_p`` as it should only access registers and is an internal function.

```
while ((now.seconds < stop.seconds) && (now.nanoseconds < stop.nanoseconds)) {
    if (!is_waiting_for_bus_idle(regs_p->STATUS)) {
        return (0);
    }
    
    time_get(&now);
}

return (-ENETDOWN);
```